src/generate_database_sqdt/main.py

```diff
@@ -0,0 +1,233 @@
+    )
+    parser.add_argument(
+        "--overwrite",
+        action="store_true",
+        help="Delete the species folder if it exists and create a new one.",
+    )
+
+    args = parser.parse_args()
+    species_folder = Path(args.directory) / f"{args.species}_v{__version__}"
+    if species_folder.exists():
+        if args.overwrite:
+            shutil.rmtree(species_folder)
+        else:
+            raise FileExistsError(f"The folder {species_folder} already exists. Use --overwrite to delete it.")
+    species_folder.mkdir(parents=True, exist_ok=True)
+    os.chdir(species_folder)
+
+    configure_logging(args.log_level, args.species)
+    time_start = time.perf_counter()
+    create_database_one_species(args.species, args.n_max)
+    logger.info("Time taken: %.2f seconds", time.perf_counter() - time_start)
+
+
+def configure_logging(log_level: str, species: str) -> None:
+    """Initialize the logger."""
+    root_logger = logging.getLogger()
+    if root_logger.hasHandlers():
+        root_logger.handlers.clear()
+    root_logger.setLevel(getattr(logging, log_level.upper()))
+
+    stream_formatter = logging.Formatter("%(levelname)s %(asctime)s: %(message)s", datefmt="%H:%M:%S")
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(stream_formatter)
+    root_logger.addHandler(stream_handler)
+
+    file_formatter = logging.Formatter("%(levelname)s: %(message)s")
+    log_file = Path(f"{species}_v{__version__}.log")
+    log_file.parent.mkdir(parents=True, exist_ok=True)
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setFormatter(file_formatter)
+    root_logger.addHandler(file_handler)
+
+
+def create_database_one_species(species: str, n_max: int) -> None:
+    """Create database for a given species."""
+    logger.info("Start creating database for %s and version v%s", species, __version__)
+    logger.info("n-max=%d", n_max)
+    logger.info("ryd_numerov.__version__=%s", ryd_numerov.__version__)
+
+    db_file = Path("database.db")
+    with sqlite3.connect(db_file) as conn:
+        conn.executescript((Path(__file__).parent / "database.sql").read_text(encoding="utf-8"))
+        list_of_states = get_sorted_list_of_states(species, n_max)
+        populate_states_table(list_of_states, conn)
+        populate_matrix_elements_table(list_of_states, conn)
+    logger.info("Size of %s: %.6f megabytes", db_file, db_file.stat().st_size * 1e-6)
+
+    with sqlite3.connect(db_file) as conn:
+        for tkey in ["states", *MATRIX_ELEMENTS_OF_INTEREST.keys()]:
+            parquet_file = Path(f"{tkey}.parquet")
+            table = pd.read_sql_query(f"SELECT * FROM {tkey}", conn)
+            if tkey == "states":
+                table = table.astype({"is_j_total_momentum": bool, "is_calculated_with_mqdt": bool})
+                table["is_j_total_momentum"] = True
+                table["is_calculated_with_mqdt"] = False
+            table.to_parquet(parquet_file, index=False)
```

Comment:
Can a compression algorithm be chosen? Some time ago, I played around and "ZSTD" seemed to be a good compromise between speed and file size.